jujugui/static/gui/src/app/components/shared/status/table/table.js

```diff
@@ -0,0 +1,64 @@
+/* Copyright (C) 2018 Canonical Ltd. */
+'use strict';
+
+const PropTypes = require('prop-types');
+const React = require('react');
+
+const BasicTable = require('../../../basic-table/basic-table');
+
+class StatusTable extends React.Component {
+  /**
+    A function to be supplied to sort methods to sort by key.
+    @param {String} a The first value.
+    @param {String} b The second value.
+    @returns {Int} The position change.
+  */
+  _byKey(a, b) {
+    if (a.key < b.key) {
+      return -1;
+    }
+    if (a.key > b.key) {
+      return 1;
+    }
+    return 0;
+  }
+
+  /**
+    Filter a row by the status.
+    @param row {Object} The row values.
+    @returns {Boolean} Whether the row matches the status.
+  */
+  _filterByStatus(row) {
+    if (!this.props.statusFilter) {
+      return true;
+    }
+    return row.extraData === this.props.statusFilter;
```

Comment:
Should we specify a more complex object for `extraData` so that this comparison makes more sense?
```javascript
return row.extraData.filter === this.props.statusFilter
```